### PR TITLE
MPI_Abort scope

### PR DESCRIPTION
**Pull request by raffenet**

### CODE REVIEW

**Comment by raffenet** [comment]
test:jenkins/ch3/tcp

**Comment by raffenet** [comment]
test:jenkins/ch3/tcp

File: src/mpi/init/abort.c [hunk 1]
```diff
@@ -121,8 +121,8 @@ int MPI_Abort(MPI_Comm comm, int errorcode)
     
     if (!comm_ptr)
     {
-	/* Use comm world if the communicator is not valid */
-	comm_ptr = MPIR_Process.comm_world;
+	/* Use comm self if the communicator is not valid */
+	comm_ptr = MPIR_Process.comm_self;
```
Comment (wesbland) [inline src/mpi/init/abort.c]: I think if you pass in an invalid communicator, the function should return `MPI_ERR_ARG` and not abort anything.

File: src/mpid/ch3/src/mpid_abort.c [hunk 2]
```diff
@@ -83,10 +83,11 @@ int MPID_Abort(MPIR_Comm * comm, int mpi_errno, int exit_code,
...
     /* FIXME: What is the scope for PMI_Abort?  Shouldn't it be one or more
-       process groups?  Shouldn't abort of a communicator abort either the
-       process groups of the communicator or only the current process?
-       Should PMI_Abort have a parameter for which of these two cases to
-       perform? */
+       process groups?  Shouldn't abort of a communicator abort the process
+       group of the communicator? Should PMI_Abort have a parameter for
+       which of these two cases to perform? */
+    if (MPIR_Comm_size(comm) == 1)
+        MPL_exit(exit_code);
```
Comment (wesbland) [inline src/mpid/ch3/src/mpid_abort.c]: At least the scope of uncertainty was narrowed somewhat. 😄

File: src/mpi/init/abort.c (hunk 1 above)
Comment (raffenet) [reply src/mpi/init/abort.c]: Hmm, I'd be reluctant to not abort when the user asks. It just so happens this is not as foolproof as `abort()`, but IMO, if it compiles, it should cause an abort.

Comment (wesbland) [reply src/mpi/init/abort.c]: If they don’t change the error handler, they’re still going to abort. 😀

Comment (raffenet) [reply src/mpi/init/abort.c]: True. I guess what I'm saying is that if a program compiles and reaches `MPI_Abort`, we should at minimum abort that process no matter what. Whatever messages/return codes the user gets may not make sense, but continuing to run seems like a worse option.

**Comment by yfguo** [comment]
test:jenkins/ch3/tcp

**Comment by yfguo** [comment]
test:jenkins/ch3/tcp

**Comment by yfguo** [comment]
test:jenkins/ch3/tcp

**Comment by yfguo** [comment]
rebased and removed the changes to CH3. Will merge if the test shows up ok.

**Pull request merged**